src/schema/tree.rs

```diff
@@ -115,6 +122,21 @@ impl Tree {
 
         None
     }
+
+    pub fn get_revision_by_hash(
+        &self,
+        hash_par: HexString<Vec<u8>>,
+    ) -> std::option::Option<AnyRevision> {
+        let mut revision_result: Option<AnyRevision> = None;
+
+        for (hash, revision) in &self.revisions {
+            if *hash == hash_par {
+                revision_result = Some(revision.clone());
+                break;
+            }
+        }
+        revision_result
+    }
```

Comment:
Can't you just use `self.revisions.get(hash_par)`?
```suggestion

    pub fn get_revision_by_hash(
        &self,
        hash_par: &RevisionLink,
    ) -> Option<&AnyRevision> {
        self.revisions.get(hash_par).as_ref()
    }
```